code/__HELPERS/unsorted.dm

```diff
@@ -1035,19 +1033,29 @@ proc/DuplicateObject(obj/original, var/perfectcopy = 0 , var/sameloc = 0)
 				var/datum/coords/C_trg = refined_trg[B]
 				if(C_src.x_pos == C_trg.x_pos && C_src.y_pos == C_trg.y_pos)
 
+					var/old_name = T.name
 					var/old_dir1 = T.dir
 					var/old_icon_state1 = T.icon_state
 					var/old_icon1 = T.icon
 
+					var/datum/gas_mixture/new_air
+					if(T.air)
+						new_air = new
+						new_air.copy_from(T.air)
```

Comment:
This should use `T.return_air()` instead of just `T.air`. Also personally I wouldn't rely on the fact that the default volume for a `gas_mixture` is `CELL_VOLUME` and would manually set it for `new_air` before copying the air to it, but that's just a nitpick. Perhaps someday I'll make `gas_mixture/New()` accept different sets of args for different purposes so this stuff is simpler.
Also I don't believe there is any circumstance in which `return_air()` can return null, so you can skip the check if you change it to that.